### PR TITLE
Allow search of org by org name

### PR DESCRIPTION
**Pull request by jonnonz**
- Bugfixes, ensure we return a pointer result in org create and update.
- Allow for user to search org by name.

### CODE REVIEW

**Comment by jonnonz** [comment]
@pzurek Mind taking a look?

**Comment by pzurek** [comment]
I'm a bit torn between accepting this or creating a separate new search service since it operates on a different endpoint. Let's chat tomorrow.

File: organization.go [hunk 1]
```diff
@@ -4,10 +4,20 @@ import (
...
+}
+
+// OrganizationWrapper struct
 type OrganizationWrapper struct {
 	Organization *Organization `   json:"organization"`
 }
 
+// OrganizationResponse struct
 type OrganizationResponse struct {
 	Organizations Organization `json:"organizations,omitempty"`
```
Comment (pzurek) [inline organization.go]: That doesn't seem OK. Shouldn't that be `Organizations []Organization 'json:"organizations,omitempty"'`?
If it should we should probably settle on the same naming as for ticket collections.

**Comment by pzurek** [comment]
Hey @jonnonz I'm merging this but hold off on pulling the library for a few minutes. I'll refactor some method names quickly to make them consistent.

**Pull request merged**